lib_eio_linux/eio_linux.ml

```diff
@@ -110,6 +111,36 @@ module FD = struct
     match t.fd with
     | `Open fd -> Fmt.pf f "%d" (Obj.magic fd : int)
     | `Closed -> Fmt.string f "(closed)"
+
+  let fstat t =
+    (* todo: use uring  *)
+    let ust = Unix.LargeFile.fstat (get_exn "fstat" t) in
+    let st_kind : Eio.File.Stat.kind =
+      match ust.st_kind with
+      | Unix.S_REG  -> `Regular_file
+      | Unix.S_DIR  -> `Directory
+      | Unix.S_CHR  -> `Character_special
+      | Unix.S_BLK  -> `Block_device
+      | Unix.S_LNK  -> `Symbolic_link
+      | Unix.S_FIFO -> `Fifo
+      | Unix.S_SOCK -> `Socket
+    in
+    Eio.File.Stat.{
+      dev     = ust.st_dev   |> Int64.of_int;
+      ino     = ust.st_ino   |> Int64.of_int;
+      kind    = st_kind;
+      perm    = ust.st_perm;
+      nlink   = ust.st_nlink |> Int64.of_int;
+      uid     = ust.st_uid   |> Int64.of_int;
+      gid     = ust.st_gid   |> Int64.of_int;
+      rdev    = ust.st_rdev  |> Int64.of_int;
+      size    = ust.st_size  |> Optint.Int63.of_int64;
+      blksize = 512          |> Int64.of_int;         (* XXX *)
+      blocks  = Int64.(div ust.st_size (of_int 512)); (* XXX *)
```

Comment:
What's the plan for dealing with these?

Reply:
I think we should just zap them, I don't see us adding stubs just to fetch those. When I added the Luv would expose the real ones at least.